language/move-vm/types/src/values/values_impl.rs

```diff
@@ -191,11 +191,21 @@ pub struct Struct {
 ///   enforces mutability restrictions.
 /// * `move_from` is represented through an owned `GlobalValue`.
 #[derive(Debug)]
-pub struct GlobalValue {
-    status: Rc<RefCell<GlobalDataStatus>>,
-    fields: Rc<RefCell<Vec<ValueImpl>>>,
+enum GlobalValueImpl {
+    None,
+    Fresh {
+        fields: Rc<RefCell<Vec<ValueImpl>>>,
+    },
+    Cached {
+        fields: Rc<RefCell<Vec<ValueImpl>>>,
+        status: Rc<RefCell<GlobalDataStatus>>,
+    },
+    Deleted,
```

Comment:
Reviewers: please pay attention to the states encoded by this enum and GlobalDataStatus, and the transition rules via the operations below. These are the core and tricky part of this PR.